lib/jekyll-import/importers/rss.rb

```diff
@@ -52,14 +55,24 @@ def self.write_rss_item(item, options)
         post_name = Jekyll::Utils.slugify(item.title, :mode => "latin")
         name = "#{formatted_date}-#{post_name}"
         audio = render_audio && item.enclosure.url
+        canonical_link = options.fetch("canonical_link", false)
 
         header = {
           "layout" => "post",
-          "title"  => item.title,
-        }
+          "title" => item.title,
+          "canonical_url" => (canonical_link ? item.link : nil),
+        }.compact
 
         header["tag"] = options["tag"] unless options["tag"].nil? || options["tag"].empty?
 
+        if options["extract_tags"]
+          tags_from_feed = item.instance_variable_get("@#{options["extract_tags"]}")
+          unless tags_from_feed.nil?
+            tags = tags_from_feed.map { |feed_tag| feed_tag.content.downcase }
+            header["tag"] = tags unless tags.empty?
+          end
+        end
```

Comment:
This section of code is confusing.
- First we check for `options["extract_tags"]` where `options` is the CLI options.
- Then we check for `item.instance_variable_get("@#{options["extract_tags"]}")` using same `options` object. So why not store the `key` in a variable for reuse?
- Then we preprocess feed-tags using `#map`. That means we expect an array of `feed_tag` instead of *any non-nil object*. So, we should explicitly test for array instead. We also *downcase* feed_tags. So, if the feed originally contained `["holiday", "Holiday"]`, we'd end up with: `["holiday", "holiday"]`. So we should filter duplicates.

```suggestion
        tag_name = options["extract_tags"]
        if tag_name
          tags_from_feed = item.instance_variable_get("@#{tag_name}")
          if tags_from_feed.is_a?(Array)
            tags = tags_from_feed.map { |feed_tag| feed_tag.content.downcase }
            tags.uniq!
            header["tag"] = tags unless tags.empty?
          end
        end
```

Reply:
Sure, this code looks much better :). Only thing `tag_name` is actually the `key` mentioned by CLI options. Therefor I'll update it accordingly.

Reply:
I know that CLI option param `KEY` corresponds to the `tag_name` in my suggestion. But now that you mentioned it, I wonder why the option is in plural `extract_tags` and why not singular as in
`--extract_tag TAG_NAME`..?

Reply:
Hi @ashmaroli, the reason of introducing `--extract_tags` is to extract tag(s) from RSS feed. When someone add a post they'll have multiple tags. In this case, you'll provide the **key**(sub field of RSS payload) and it'll add all those tags to frontmatter. 

For example, if you look into the follow RSS feed from Medium, it has multiple **category** (which is actually tags).

```xml
<?xml version="1.0" encoding="UTF-8"?>
<rss xmlns:dc="http://purl.org/dc/elements/1.1/" xmlns:content="http://purl.org/rss/1.0/modules/content/"
     xmlns:atom="http://www.w3.org/2005/Atom" version="2.0"
     xmlns:cc="http://cyber.law.harvard.edu/rss/creativeCommonsRssModule.html">
    <channel>
        <title><![CDATA[Stories by Suman Maity on Medium]]></title>
        <description><![CDATA[Stories by Suman Maity on Medium]]></description>
        <link>https://medium.com/@suman.maity112?source=rss-92eeb3145de9------2</link>
        <image>
            <url>https://cdn-images-1.medium.com/fit/c/150/150/1*rAk1Q7-7kFOM-arRqtpvMA.jpeg</url>
            <title>Stories by Suman Maity on Medium</title>
            <link>https://medium.com/@suman.maity112?source=rss-92eeb3145de9------2</link>
        </image>
        <generator>Medium</generator>
        <lastBuildDate>Mon, 07 Nov 2022 19:38:06 GMT</lastBuildDate>
        <atom:link href="https://medium.com/@suman.maity112/feed" rel="self" type="application/rss+xml"/>
        <webMaster><![CDATA[yourfriends@medium.com]]></webMaster>
        <atom:link href="http://medium.superfeedr.com" rel="hub"/>
        <item>
            <title><![CDATA[Is it the era of mono-repo?]]></title>
            <link>
                https://medium.com/@suman.maity112/is-it-the-era-of-mono-repo-671e6dee387?source=rss-92eeb3145de9------2
            </link>
            <guid isPermaLink="false">https://medium.com/p/671e6dee387</guid>
            <category><![CDATA[version-control]]></category>
            <category><![CDATA[multi-repo]]></category>
            <category><![CDATA[monorepo]]></category>
            <dc:creator><![CDATA[Suman Maity]]></dc:creator>
            <pubDate>Mon, 07 Nov 2022 08:11:46 GMT</pubDate>
            <atom:updated>2022-11-07T08:11:46.975Z</atom:updated>
            <content:encoded><![CDATA[dummy content]]></content:encoded>
        </item>
    </channel>
</rss>```

This is the main reason of having `--extract_tags` as plural instead of singular. You can find the imported RSS feed [here](https://sumanmaity112.github.io/posts/is-it-the-era-of-mono-repo/)